File: src/test/java/webdriver/elements/Table.java
package webdriver.elements;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class Table extends BaseElement{

    public Table(final By locator, final String name) {
        super(locator, name);
    }

    public Table(String string, String name) {
        super(string, name);
    }

    public Table(By locator) {
        super(locator);
    }

    @Override
    protected String getElementType() {
        return "Table";
    }

    public int getRowCount() {
        return getElement().findElements(By.tagName("tr")).size();
    }

    public WebElement getRow(int row) {
        return getElement().findElements(By.tagName("tr")).get(row);
    }

    public int getColumnCount() {
        return getRow(0).findElements(By.tagName("td")).size();
    }

    public void clickCell(int row, int column) {
        WebElement cell = getRow(row).findElements(By.tagName("td")).get(column);
        cell.click();
    }

    public String getCellClassName(int row, int column) {
        return getRow(row).findElements(By.tagName("td")).get(column).getAttribute("class");
    }
}

File: TestDeveloper/SmartFramework/framework/Elements/Table.cs
using System.Collections.Generic;
using OpenQA.Selenium;

namespace smart.framework.Elements
{
    public class Table : BaseElement
    {
        public Table(By locator, string name) : base(locator, name)
        {
        }

        private IWebElement GetRow(int row)
        {
            return GetElement().FindElements(By.TagName("tr"))[row];
        }

        private IWebElement GetCellElement(Cell cell)
        {
            return GetRow(cell.Row).FindElements(By.TagName("td"))[cell.Column];
        }

        public int GetRowCount()
        {
            return GetElement().FindElements(By.TagName("tr")).Count;
        }

        public int GetColumnCount()
        {
            return GetRow(0).FindElements(By.TagName("td")).Count;
        }

        public string GetCellText(Cell cell, By childElement = null) => childElement == null ? GetCellElement(cell).Text : GetCellElement(cell).FindElement(childElement).Text;

        public List<string> GetColumnText(int column, int firstRow = 0, By childElement = null)
        {
            var result = new List<string>();

            for (var i = firstRow; i < GetColumnCount(); i++)
            {
                result.Add(GetCellText(new Cell(firstRow, column), childElement));
            }

            return result;
        }
    }
}
